File: src/features/map/types/coordinate.ts
/** Ponto geográfico em graus decimais (WGS 84). */
export type Coordinate = {
  latitude: number;
  longitude: number;
};

/** Coordenada com um rótulo legível, usada para origem e destino. */
export type NamedCoordinate = Coordinate & {
  name: string;
};

File: backend/app/schemas/coordinate.py
"""Ponto geográfico (latitude e longitude). Igual ao tipo `Coordinate` do app."""

from pydantic import BaseModel, ConfigDict, Field


class Coordinate(BaseModel):
    """Coordenada em graus decimais (padrão WGS 84)."""

    model_config = ConfigDict(frozen=True)

    latitude: float = Field(ge=-90, le=90)
    longitude: float = Field(ge=-180, le=180)
